src/rust/ensogl/src/display/scene.rs

```diff
@@ -336,18 +340,21 @@ impl Mouse {
         let mouse_manager   = MouseManager::new(&document.into());
         let frp             = frp::io::Mouse::new();
 
-        let on_move = mouse_manager.on_move.add(f!([frp,shape,position,last_position](event:&mouse::OnMove) {
-            let pixel_ratio  = shape.value().pixel_ratio as i32;
-            let screen_x     = event.client_x();
-            let screen_y     = event.client_y();
+        let on_move = mouse_manager.on_move.add(f!([frp,scene_frp,position,last_position]
+        (event:&mouse::OnMove) {
+            let shape       = scene_frp.shape.value();
+            let pixel_ratio = shape.pixel_ratio as i32;
+            let screen_x    = event.client_x();
+            let screen_y    = event.client_y();
 
             let new_position = Vector2::new(screen_x,screen_y);
             let pos_changed  = new_position != last_position.get();
             if pos_changed {
                 last_position.set(new_position);
                 let new_canvas_position = new_position * pixel_ratio;
                 position.set(new_canvas_position);
-                let position = enso_frp::Position::new(new_position.x as f32,new_position.y as f32);
+                let center   = Vector2(shape.width/2.0,shape.height/2.0);
+                let position = Vector2(new_position.x as f32,new_position.y as f32) - center;
```

Comment:
This calls for some utilities, like having a type for "Rect" with a `center` method.